handlers/handlers.go

```diff
@@ -121,3 +221,79 @@ func getRequestBody(request *http.Request) []byte {
 
 	return body
 }
+
+
+// Token Util
+const oneWeekInMinutes = 60 * 24 * 7
+const cerealNotesCookieName = "CerealNotesToken"
+
+func parseTokenFromString(tokenString string) (*jwt.Token, error) {
+	// Parse the token
+	token, err := jwt.ParseWithClaims(
+		strings.TrimSpace(tokenString),
+		&CerealNotesClaims{},
+		func(token *jwt.Token) (interface{}, error) {
+			return tokenSigningKey, nil
+		})
+	return token, err
+}
+
+func createTokenAsString(
+	userId models.UserId,
+	expirationTimeInMinutes int64,
+) (string, error) {
+	// Create the Claims
+	claims := CerealNotesClaims{
+		userId,
+		jwt.StandardClaims{
+			ExpiresAt: time.Now().Unix() + (expirationTimeInMinutes * 60),
+			Issuer:    "CerealNotes",
+		},
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	ss, err := token.SignedString(tokenSigningKey)
+	return ss, err
+}
+
+func getUserIdFromStoredToken(request *http.Request) (models.UserId, error) {
+	cookie, err := request.Cookie(cerealNotesCookieName)
+	if err != nil {
+		return 0, err
+	}
+
+	token, err := parseTokenFromString(cookie.Value)
+	if err != nil {
+		return 0, err
+	}
+
+	if claims, ok := token.Claims.(*CerealNotesClaims); ok && token.Valid {
+		return claims.UserId, nil
+	} else {
+		return 0, errors.Errorf("Token was invalid or unreadable")
+	}
+}
+
+func tokenTest1() {
```

Comment:
I agree. I was waiting for you to implement the test framework so i could just throw it into whatever you made. Instead of just arbitrarily adding names or files now. I'll add a subtask to implement test framework to move this function.